pkg/pillar/cmd/zedrouter/dnsmasq.go

```diff
@@ -257,45 +294,46 @@ func createDnsmasqConfiglet(
 		if advertizeRouter && !ctx.disableDHCPAllOnesNetMask {
 			// Network prefix "255.255.255.255" will force packets to go through
 			// dom0 virtual router that makes the packets pass through ACLs and flow log.
-			file.WriteString(fmt.Sprintf("dhcp-option=option:netmask,%s\n",
+			buffer.WriteString(fmt.Sprintf("dhcp-option=option:netmask,%s\n",
 				"255.255.255.255"))
 		} else {
-			file.WriteString(fmt.Sprintf("dhcp-option=option:netmask,%s\n",
+			buffer.WriteString(fmt.Sprintf("dhcp-option=option:netmask,%s\n",
 				ipv4Netmask))
 		}
 	}
 	if advertizeRouter {
 		// IPv6 XXX needs to be handled in radvd
 		if !isIPv6 {
-			file.WriteString(fmt.Sprintf("dhcp-option=option:router,%s\n",
+			buffer.WriteString(fmt.Sprintf("dhcp-option=option:router,%s\n",
 				router))
 			if !ctx.disableDHCPAllOnesNetMask {
-				file.WriteString(fmt.Sprintf("dhcp-option=option:classless-static-route,%s/32,%s,%s,%s,%s,%s\n",
+				buffer.WriteString(fmt.Sprintf("dhcp-option=option:classless-static-route,%s/32,%s,%s,%s,%s,%s\n",
 					router, "0.0.0.0",
 					"0.0.0.0/0", router,
 					netstatus.Subnet.String(), router))
 			}
 		}
 	} else {
-		log.Functionf("createDnsmasqConfiglet: no router\n")
+		log.Functionf("createDnsmasqConfigletWithBuffer: no router\n")
 		if !isIPv6 {
-			file.WriteString(fmt.Sprintf("dhcp-option=option:router\n"))
+			buffer.WriteString(fmt.Sprintf("dhcp-option=option:router\n"))
 		}
 		if !advertizeDns {
 			// Handle isolated network by making sure
 			// we are not a DNS server. Can be overridden
 			// with the DnsServers above
-			log.Functionf("createDnsmasqConfiglet: no DNS server\n")
-			file.WriteString(fmt.Sprintf("dhcp-option=option:dns-server\n"))
+			log.Functionf("createDnsmasqConfigletWithBuffer: no DNS server\n")
+			buffer.WriteString(fmt.Sprintf("dhcp-option=option:dns-server\n"))
 		}
 	}
-	if netstatus.DhcpRange.Start != nil {
-		dhcpRange = netstatus.DhcpRange.Start.String()
-	}
 	if isIPv6 {
-		file.WriteString(fmt.Sprintf("dhcp-range=::,static,0,60m\n"))
+		buffer.WriteString("dhcp-range=::static,0,60m\n")
 	} else {
-		file.WriteString(fmt.Sprintf("dhcp-range=%s,static,%s,60m\n",
+		dhcpRange, err := dhcpRangeConfigLine(netstatus.DhcpRange.Start, netstatus.DhcpRange.End)
```

Comment:
What about renaming to `dhcpv4RangeConfigLine` just to make it clear that it would not work for IPv6?